src/init/features/dataconnect/index.ts

```diff
@@ -52,29 +53,60 @@ export async function doSetup(setup: Setup, config: Config): Promise<void> {
     type: "input",
     default: "my-connector",
   });
-  const dir: string =
-    config.get("dataconnect.source") ||
-    (await promptOnce({
-      message: "What directory should be used for DataConnect config and schema?",
-      type: "input",
-      default: "dataconnect",
-    }));
+  const dir: string = config.get("dataconnect.source") || "dataconnect";
   if (!config.has("dataconnect")) {
     config.set("dataconnect.source", dir);
     config.set("dataconnect.location", locationId);
   }
-  // TODO: Listinstances from CloudSQL, and filter to only the free trial appropriate ones.
-  const cloudSqlInstanceId = await promptOnce({
-    message: `What CloudSQL instance ID in ${locationId} would you like to use (it will be created if it does not exist)?`,
-    type: "input",
-    default: `dataconnect-test`,
-  });
-  // TODO: ListDatabases from CloudSQL.
-  const cloudSqlDatabase = await promptOnce({
-    message: `Which database would you like to use from ${cloudSqlInstanceId} (it will be created if it does not exist)?`,
-    type: "input",
-    default: `dataconnect-test`,
-  });
+  let cloudSqlInstanceId = "";
+  let newInstance = false;
+  if (setup.projectId) {
+    const instances = await cloudsql.listInstances(setup.projectId);
+    const instancesInLocation = instances.filter((i) => i.region === locationId);
+    const choices = instancesInLocation.map((i) => {
+      return { name: i.name, value: i.name };
+    });
+    choices.push({ name: "Create a new instance", value: "" });
+    if (instancesInLocation.length) {
+      cloudSqlInstanceId = await promptOnce({
+        message: `Which CloudSSQL in ${locationId} would you like to use?`,
+        type: "list",
+        choices,
+      });
+    }
+  }
+  if (cloudSqlInstanceId === "") {
+    newInstance = true;
+    cloudSqlInstanceId = await promptOnce({
+      message: `What ID would you like to use for your new CloudSQL instance?`,
+      type: "input",
+      default: `dataconnect-test`,
+    });
+  }
```

Comment:
Playing around with the CLI a bit more. Just an idea for consideration.

For customers that just finished console onboard, they already have a CSQL instance created, would be it nice ask them to pick an existing cloud sql instance as the first question?

If they picked an existing instance, then no need to ask about region. Basically. Right now, need to remember the region of my instance in order to pick it in CLI.